kobo/apps/long_running_migrations/jobs/0007_backfill_attachment_model.py

```diff
@@ -1,61 +1,74 @@
+            for attachment in attachments:
+                instance = attachment.instance
+
+                attachment.xform_id =  instance.xform_id
+                attachment.user_id = instance.xform.user_id
+
+                if not attachment.uid:
+                    attachment.uid = KpiUidField.generate_unique_id('att')
+
+                if not attachment.date_created:
+                    attachment.date_created = instance.date_created
+
+                if not attachment.date_modified:
+                    attachment.date_modified = attachment.date_created
+
+                if attachment.deleted_at and not attachment.delete_status:
+                    attachment.date_modified = attachment.deleted_at
+                    attachment.delete_status = AttachmentDeleteStatus.SOFT_DELETED
+
+            Attachment.all_objects.bulk_update(
+                attachments,
+                fields=[
+                    'xform_id',
+                    'user_id',
+                    'uid',
+                    'date_created',
+                    'date_modified',
+                    'delete_status',
+                ],
+            )
+            attachment_pk = attachment.pk
+        else:
+            stop = True
+
+
+def get_queryset(attachment_pk):
+    return (
+        Attachment.all_objects.select_related('instance__xform')
+        .only(
+            'id',
             'deleted_at',
             'date_created',
-            'instance__date_created',
+            'date_modified',
+            'instance_id',
             'instance__xform_id',
             'instance__xform__user_id',
         )
-    ).iterator(CHUNK_SIZE)
-
-    for attachment_batch in chunked(attachments, CHUNK_SIZE):
-        for attachment in attachment_batch:
-            instance = attachment.instance
-            xform = instance.xform
-            user = xform.user
-
-            attachment.xform_id = xform.id
-            attachment.user_id = user.id
-
-            if not attachment.uid:
-                attachment.uid = KpiUidField.generate_unique_id('att')
-
-            if not attachment.date_created:
-                attachment.date_created = instance.date_created
-
-            if not attachment.date_modified:
-                attachment.date_modified = attachment.date_created
-
-            if attachment.deleted_at and not attachment.delete_status:
-                attachment.date_modified = attachment.deleted_at
-                attachment.delete_status = AttachmentDeleteStatus.SOFT_DELETED
-
-        Attachment.all_objects.bulk_update(
-            attachment_batch,
-            fields=[
-                'xform_id',
-                'user_id',
-                'uid',
-                'date_created',
-                'date_modified',
-                'delete_status',
-            ],
-        )
+        .filter(
+            Q(date_created__isnull=True) | Q(date_modified__isnull=True),
+            pk__gt=attachment_pk,
+        )[:CHUNK_SIZE]
```

Comment:
You need to `order_by('pk')` to ensure we are always fetching the correct subset of data. Right now it is random, nothing guarantee the subset will contain `CHUNK_SIZE` elements.